subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskNoTestsFailTest.groovy

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.api.tasks.testing.TestExecutionException
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.executer.UnexpectedBuildFailure
+
+class TestTaskNoTestsFailTest extends AbstractIntegrationSpec {
+
+    def "test --failIfNoTest fails if there is no test"() {
+        buildFile << "apply plugin: 'java'"
+
+        file("src/test/java/NotATest.java") << """
+            public class NotATest {}
```

Comment:
We need also coverage for when there are tests using one testing framework but the test task is configured to use another testing framework.